### ISSUE TITLE
`block-closing-brace-space-before` reads a stray semicolon behind a rule's block as that block's closing brace

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#562]
### What minimal example or steps are needed to reproduce the bug?

A stylesheet whose block is followed by a stray semicolon:

```css
a {
	b { c: d }
	;
}
```

PostCSS files the whitespace in front of such a semicolon, along with the semicolon itself, in the `raws.ownSemicolon` of the rule it follows, and its stringifier prints that raw directly behind the rule's closing brace. `postcss-scss` and `postcss-less` read and print it exactly as it does.

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": [
		"@stylistic/stylelint-plugin"
	],
	"rules": {
		"@stylistic/block-closing-brace-space-before": "always"
	}
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

The rule to ask about the character standing in front of the block's closing brace, and one run of `--fix` to settle the file.

### What actually happened?

It asks about the character standing in front of the stray semicolon instead, and writes what the option asks for into the block's `raws.after`, which is another text entirely. So the warning survives every run: the stylesheet above is reported at `3:1`, `--fix` writes `a {`, a break, a tab, `b { c: d }`, a break, a tab, `; }` — the closing brace of the outer block moved, which is a warning of its own answered correctly — and `3:1` is reported again, on that run and on every run after it.

Every option of the rule reaches it, and all three syntaxes answer alike. Twelve shapes out of the five below, run to a fixed point:

| option | input | `--fix` rests on | reported |
| --- | --- | --- | --- |
| `always` | `a {\n\tb { c: d }\n\t;\n}\n` | `a {\n\tb { c: d }\n\t; }\n` | `3:1` |
| `always` | `a{b{c:d};}\n` | `a{b{c:d }; }\n` | `1:9` |
| `always` | `a { b: c }\n;\n` | unchanged | `1:11` |
| `never` | `a {\n\tb { c: d }\n\t;\n}\n` | `a {\n\tb { c: d}\n\t;}\n` | `3:1` |
| `never` | `a { b { c: d } ; }\n` | `a { b { c: d} ;}\n` | `1:14` |
| `never` | `a { b: c }\n;\n` | `a { b: c}\n;\n` | `1:10` |
| `always-single-line` | `a{b{c:d};}\n` | `a{b{c:d }; }\n` | `1:9` |
| `never-single-line` | `a { b { c: d } ; }\n` | `a { b { c: d} ;}\n` | `1:14` |
| `always-multi-line` | `a {\n\tb { c: d }\n\t;\n}\n` | `a {\n\tb { c: d }\n\t; }\n` | `3:1` |
| `always-multi-line` | `a { b: c }\n;\n` | unchanged | `1:11` |
| `never-multi-line` | `a {\n\tb { c: d }\n\t;\n}\n` | `a {\n\tb { c: d}\n\t;}\n` | `3:1` |
| `never-multi-line` | `a { b: c }\n;\n` | `a { b: c}\n;\n` | `1:10` |

A rule of the top level is reached as readily as a nested one: the parser hands the raw to any rule a stray semicolon stands behind. A block with no such semicolon behind it is read correctly under every option.

### What is the cause?

[`blockString`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/utils/blockString/index.ts) slices the printed node, and PostCSS prints `raws.ownSemicolon` behind the closing brace, so the string it hands back ends on the semicolon rather than on the brace:

```js
let root = postcss.parse(`a {\n\tb { c: d }\n\t;\n}\n`)
blockString(root.first.first) // `{ c: d }\n\t;`
```

The rule takes the last character of that string for the closing brace and asks the checker about the character in front of it — the tab in front of the semicolon. The `source.end` of such a rule is moved the same way, `freeSemicolon` pushing it past the brace and onto the semicolon, so a reading taken from the node's own end lands there too.

### Do you have a proposal to fix the bug?

Stop the block at its closing brace: `raws.ownSemicolon` stands behind the brace and is no part of the block, so a reading of the block is a reading of the printed node with that raw taken off the end. The raw is absent from every node no stray semicolon stands behind, so the reading narrows nothing else.

### Notes

Found while fixing [#372](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/372), which is the same raw read by `linebreaks`, and measured on both sides of that branch alike, so it is older than that branch and untouched by it.

The two other rules of the family read the same string and are misled in the same way, and neither leaves a warning behind: a question about a line break, or about an empty line, happens to be answered the same way by the run in front of the semicolon as by the run in front of the brace in every shape measured. So only this rule reports what no run can take away.

It is the third shape of one failure on this family, and a different cause from the two standing already: [#538](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/538) is a run the declaration keeps and the rule writes past, [#374](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/374) is a run `postcss-less` collects into one raw on both sides of a flag, and this is the closing brace itself being read one character too far along.
